Leanplum/Leanplum.m

```diff
@@ -289,6 +338,179 @@ + (BOOL)hasStarted
     return [LPInternalState sharedState].hasStarted;
 }
 
++ (NSString *)pushTokenKey
+{
+    return [NSString stringWithFormat: LEANPLUM_DEFAULTS_PUSH_TOKEN_KEY,
+            [LPAPIConfig sharedConfig].appId, [LPAPIConfig sharedConfig].userId, [LPAPIConfig sharedConfig].deviceId];
+}
+
++ (void)start
+{
+    [self startWithUserId:nil userAttributes:nil withSuccess:nil withFailure:nil];
+}
+
++ (void)startWithResponseHandler:(LeanplumStartBlock)response
+{
+    [self startWithUserId:nil userAttributes:nil withSuccess:^{
+        response(true);
+    } withFailure:^(NSError *error){
+        response(false);
+    }];
+}
+
++ (void)startWithUserAttributes:(NSDictionary *)attributes
+{
+    [self startWithUserId:nil userAttributes:attributes withSuccess:nil withFailure:nil];
+}
+
++ (void)startWithUserId:(NSString *)userId
+{
+    [self startWithUserId:userId userAttributes:nil withSuccess:nil withFailure:nil];
+}
+
++ (void)startWithUserId:(NSString *)userId responseHandler:(LeanplumStartBlock)response
+{
+    [self startWithUserId:userId userAttributes:nil withSuccess:^{
+        response(true);
+    } withFailure:^(NSError *error){
+        response(false);
+    }];
+}
+
++ (void)startWithUserId:(NSString *)userId userAttributes:(NSDictionary *)attributes
+{
+    [self startWithUserId:userId userAttributes:attributes withSuccess:nil withFailure:nil];
+}
+
++ (void)startWithUserId:(NSString *)userId
+            userAttributes:(NSDictionary *)attributes
+            withSuccess:(void (^)(void))success
+            withFailure:(void (^)(NSError *error))failure {
+   
+    if ([LPAPIConfig sharedConfig].appId == nil) {
+        [self throwError:@"Please provide your app ID using one of the [Leanplum setAppId:] "
+         @"methods."];
+        return;
+    }
+    
+    // Leanplum should not be started in background.
+    if (![NSThread isMainThread]) {
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            [self startWithUserId:userId userAttributes:attributes withSuccess:success withFailure:failure];
+        });
+        return;
+    }
+    
+    attributes = [self validateAttributes:attributes named:@"userAttributes" allowLists:YES];
+    
+    // Set device ID.
+    NSString *deviceId = [LPAPIConfig sharedConfig].deviceId;
+    
+    LPInternalState *state = [LPInternalState sharedState];
+    //ToDo: IS_NOOP logic, need to decide the reason for this logic and if needed
+    
+    if (state.calledStart) {
+        [self throwError:@"Already called start."];
+    }
+    state.calledStart = YES;
+    // This is the device ID set when the MAC address is used on iOS 7.
+    // This is to allow apps who upgrade to the new ID to forget the old one.
+    if ([deviceId isEqualToString:@"0f607264fc6318a92b9e13c65db7cd3c"]) {
```

Comment:
do we need this check, minimum version supported is 8 ?

Reply:
Good catch. Will remove this check .

Reply:
Can we remove everything that is fallbacks for older versions that we will never be supporting?

Reply:
done